pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1924,7 +1947,11 @@ func (cf *changeFrontier) manageProtectedTimestamps(
 	// If we've identified more tables that need to be protected since this
 	// changefeed was created, it will be missing here. If so, we "migrate" it
 	// to include all the appropriate targets.
-	if targets := AllTargets(cf.spec.Feed); !makeTargetToProtect(targets).Equal(rec.Target) {
+	targets, err = AllTargets(ctx, cf.spec.Feed, cf.FlowCtx.Cfg.ExecutorConfig.(*sql.ExecutorConfig))
```

Comment:
IMO for both processors we should compute targets once in their `Start` functions and then store it as a field on the struct